packages/orchestration/src/utils/start-helper.js

```diff
@@ -199,16 +199,19 @@ harden(provideOrchestration);
  *   zone: Zone,
  *   tools: OrchestrationTools,
  * ) => Promise<R>} contractFn
+ * @param {WithOrchestrationOpts} [opts]
  * @returns {(zcf: ZCF<CT>, privateArgs: PA, baggage: Baggage) => Promise<R>} a
  *   Zoe start function
  */
 export const withOrchestration =
-  contractFn => async (zcf, privateArgs, baggage) => {
+  (contractFn, opts) => async (zcf, privateArgs, baggage) => {
+    const { marshaller, ...allOrchPowers } = privateArgs;
+    const { storageNode: _, ...requiredOrchPowers } = allOrchPowers;
     const { zone, ...tools } = provideOrchestration(
       zcf,
       baggage,
-      privateArgs,
-      privateArgs.marshaller,
+      opts?.disableStorageNode ? requiredOrchPowers : allOrchPowers,
+      marshaller,
```

Comment:
Was there original motivation to have `OrchestrationPowers` and `marshaller: Marshaller` separate?

Regardless, this could be refactored to also omit marshaller and skip the call to prepareRecorderKit if that's preferable.